.github/workflows/build_and_test.yml

```diff
@@ -51,11 +51,14 @@ jobs:
         KONG_VERSION=$(bash scripts/grep-kong-version.sh)
         major=$(echo "$KONG_VERSION" | cut -d. -f1)
         minor=$(echo "$KONG_VERSION" | cut -d. -f2)
-        # if the minor version isn't 0, use the first release of the previous minor version;
+        # if the minor version isn't 0, use the starting point of the previous minor branch;
         # otherwise just leave it empty, so later the default branch or commit will be used.
         if [ "$minor" -ne 0 ]; then
           minor=$((minor - 1))
-          echo "ref=$major.$minor.0" >> $GITHUB_OUTPUT
+          git fetch origin master
```

Comment:
Usually the starting point of the branch may have bug, I'm not sure if it's worthing to implement the following logic:

- When a release tag exists for the branch, use the latest release tag
- If we haven't released yet in the branch, fall back to the starting point

You could use the following commands to get the latest release tag:
```git fetch origin master -t
latest_version=$(git tag -l $major.$minor.*|sort -V|tail -1)
if [  -z "$latest_version" ]; then
  # get the starting point as we haven't released from this branch yet
  COMMIT_HASH=$(git merge-base origin/master origin/release/$major.$minor.x)
  echo "ref=$COMMIT_HASH" >> $GITHUB_OUTPUT
else
  # use the latest release tag from the branch
  echo "ref=" > $GITHUB_OUTPUT
fi
```

Reply:
I agree. It depends on the definition of 'previous version'—whether it refers to the last commit on the branch or the last released version. I think it's more likely referring to the last released version here.

Reply:
I will change to use this logic:
- if the first release tag(x.x.0) exists, use the tag
- otherwise fallback to the starting point

The latest release tag changes all the time. We had a [discussion](https://github.com/Kong/kong-ee/pull/9077#discussion_r1692784320) earlier that pinning to a fixed tag was the better way to go.